addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/Base64UrlDecoderUnitTest.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.predefined;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.withSettings;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.addon.encoder.EncodeDecodeOptions;
+import org.zaproxy.addon.encoder.ExtensionEncoder;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeResult;
+
+public class Base64UrlDecoderUnitTest extends ProcessorTests<Base64UrlDecoder> {
+
+    @Override
+    protected Base64UrlDecoder createProcessor() {
+        return Base64UrlDecoder.getSingleton();
+    }
+
+    Model model;
+    ExtensionLoader extensionLoader;
+    ExtensionEncoder extEnc;
+    EncodeDecodeOptions options;
+
+    @BeforeEach
+    void init() {
+        model = mock(Model.class, withSettings().defaultAnswer(CALLS_REAL_METHODS));
+        Model.setSingletonForTesting(model);
+        extensionLoader = mock(ExtensionLoader.class, withSettings().lenient());
+        extEnc = mock(ExtensionEncoder.class, withSettings().lenient());
+        given(extensionLoader.getExtension(ExtensionEncoder.class)).willReturn(extEnc);
+        Control.initSingletonForTesting(Model.getSingleton(), extensionLoader);
+        options = mock(EncodeDecodeOptions.class);
+        given(extEnc.getOptions()).willReturn(options);
+        given(options.getBase64Charset()).willReturn(EncodeDecodeOptions.DEFAULT_CHARSET);
+    }
+
+    @Test
+    void shouldErrorOnStringInput() throws Exception {
+        // Given / When
+        EncodeDecodeResult result = processor.process("admin");
+        // Then
+        assertThat(result.hasError(), is(equalTo(true)));
+        assertThat(
+                result.getResult(),
+                is(equalTo("IllegalArgumentException: Last unit does not have enough valid bits")));
+    }
+
+    @Test
+    void shouldHandleEncodedInput() throws Exception {
+        // Given / When
+        EncodeDecodeResult result = processor.process("YWRtaW4=");
+        // Then
+        assertThat(result.hasError(), is(equalTo(false)));
+        assertThat(result.getResult(), is(equalTo("admin")));
+    }
+
+    @Test
+    void shouldDecodeStringWhichDoesntMatchStandardBase64() throws Exception {
```

Comment:
What's not standard?

Reply:
Shall I just drop the word standard? I was trying to imply that there's a difference Base64 vs Base64Url

Reply:
`shouldDecodeBase64UrlString`? if I understood correctly.